tests/test_metadata.py

```diff
@@ -1226,28 +1226,51 @@ def assert_registered(full):
 
     def test_apply_identifier_equivalency(self):
 
-        # Set up primary identifier with matching & new IdentifierData objects
+        # Set up an Edition.
```

Comment:
The behavior of this test changed when I first implemented `Identifier.__eq__`. I didn't have the type check on `other`, so an Identifier and an IdentifierData could have Python equality if they represented the same conceptual identifier. This meant that if you mentioned an `Identifier` and an `IdentifierData` in the `Metadata` constructor, you might end up with only one or the other in `Metadata.identifiers`.

I think a more expansive implementation if `Identifier.__eq__` might be a good idea, but it's way beyond the scope of this project. But by the time I'd figured out what was going on, I'd expanded this test and made it easier to read, so I'm leaving that bit in place.